eng/versioning/set_versions.py

```diff
@@ -51,12 +51,76 @@
 version_regex_named = re.compile(version_regex_str_with_names_anchored)
 prerelease_regex_named = re.compile(prerelease_version_regex_with_name)
 
+# Update packages (excluding unreleased dependencies and packages which already
+# have a dev version set) to use a "zero dev version" (e.g. dev.20201225.0).
+# This ensures that packages built in pipelines who have unreleased dependencies
+# that are built in other pipelines can successfully fall back to a source build
+# of the unreleased dependency package in the monorepo if the unreleased
+# dependency has not been published to the dev feed yet. This makes it so
+# nightly pipelines do not have to coordinate to publish packages to the dev
+# feed in dependency order.
+# This function assumes that the version file has already been updated with dev
```

Comment:
Any reason we did this as a separate step as opposed to as part of the set dev version call?